include/openssl/base.h

```diff
@@ -182,6 +182,22 @@ extern "C" {
 
 #endif
 
+#if defined(AWSLC_DEBUG_BUILD)
+
+// AWSLC_NOOP is used to mark a no-op symbol in AWS-LC as deprecated
+// in debug builds. This allows potential reliance on these no-ops symbols
+// to be more discoverable when migrating to AWS-LC, but does not block
+// release builds that may define "-Werror".
+// Refer to "docs/porting/functionality-differences.md" for more details
+// on no-op symbols in AWS-LC.
+#define AWSLC_NOOP __attribute__((__deprecated__))
```

Comment:
You could use `OPENSSL_DEPRECATED` here and leverage the logic above this which determines the appropriate way to indicate that for the current environment.